src/VisualStudio/Core/Test.Next/Services/VisualStudioDiagnosticAnalyzerExecutorTests.cs

```diff
@@ -120,6 +123,28 @@ End Sub
             }
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.RemoteHost)]
+        [WorkItem(26178, "https://github.com/dotnet/roslyn/pull/26178")]
+        public async Task TestCancellationOnSessionWithSolution()
+        {
+            var code = @"class Test { void Method() { } }";
+
+            using (var workspace = CreateWorkspace(LanguageNames.CSharp, code))
+            {
+                var solution = workspace.CurrentSolution;
+                var solutionChecksum = await solution.State.GetChecksumAsync(CancellationToken.None);
+
+                var source = new CancellationTokenSource();
+                var connection = new InvokeThrowsCancellationConnection(source);
+                await Assert.ThrowsAnyAsync<OperationCanceledException>(() => SessionWithSolution.CreateAsync(connection, solution, source.Token));
```

Comment:
Nit: assert that the resulting exception (which is returned so you can assert with it) has OperationCanceledException.CancellationToken equal to source.CancellationToken. Since there's some catching/rethrowing going on we don't want to misassociate the cancellation source.

Reply:
> is there Assert.ThrowXXX that let me get to the exception

Yes, they all return the exception instance.